File: Release/WindowsPhone71/PhoneToolkitSample/Data/AccentColorNameToBrush.cs
// (c) Copyright Microsoft Corporation.
// This source is subject to the Microsoft Public License (Ms-PL).
// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
// All other rights reserved.

using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.Windows.Data;
using System.Windows.Media;

namespace PhoneToolkitSample.Data
{
    /// <summary>
    /// A converter that takes a name of an accent color and returns a SolidColorBrush.
    /// </summary>
    public class AccentColorNameToBrush : IValueConverter
    {
        private static Dictionary<string, SolidColorBrush> ColorNameToBrush = new Dictionary<string, SolidColorBrush>()
        {
            { "magenta", 0xFFFF0097.ToSolidColorBrush() },
            { "purple",  0xFFA200FF.ToSolidColorBrush() },
            { "teal",    0xFF00ABA9.ToSolidColorBrush() },
            { "lime",    0xFF8CBF26.ToSolidColorBrush() },
            { "brown",   0xFFA05000.ToSolidColorBrush() },
            { "pink",    0xFFE671B8.ToSolidColorBrush() },
            { "orange",  0xFFF09609.ToSolidColorBrush() },
            { "blue",    0xFF1BA1E2.ToSolidColorBrush() },
            { "red",     0xFFE51400.ToSolidColorBrush() },
            { "green",   0xFF339933.ToSolidColorBrush() },
            { "mango",   0xFFF09609.ToSolidColorBrush() },
        };

        /// <summary>
        /// Converts a name of an accent color to a SolidColorBrush.
        /// </summary>
        /// <param name="value">The accent color as a string.</param>
        /// <param name="targetType">The target type</param>
        /// <param name="parameter">The parameter</param>
        /// <param name="culture">The culture</param>
        /// <returns>A SolidColorBrush representing the accent color.</returns>
        [SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase", Justification = "By design")]
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            string v = value as string;
            if (null == v)
            {
                throw new ArgumentNullException("value");
            }

            SolidColorBrush brush = null;
            if (ColorNameToBrush.TryGetValue(v.ToLowerInvariant(), out brush))
            {
                return brush;
            }

            return null;
        }

        /// <summary>
        /// Not Implemented
        /// </summary>
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            throw new NotImplementedException();
        }
    }
}

File: Release/WindowsPhone71/PhoneToolkitSample/Data/AccentColorNameToBrush.vb
' (c) Copyright Microsoft Corporation.
' This source is subject to the Microsoft Public License (Ms-PL).
' Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
' All other rights reserved.

Imports System.Windows.Data

Namespace Data

    Public Class AccentColorNameToBrush
        Implements IValueConverter

        Private Shared ColorNameToBrush As Dictionary(Of String, SolidColorBrush)

        Shared Sub New()
            ColorNameToBrush.Add("magenta", &HFFFF0097UI.ToSolidColorBrush())
            ColorNameToBrush.Add("purple", &HFFA200FFUI.ToSolidColorBrush())
            ColorNameToBrush.Add("teal", &HFF00ABA9UI.ToSolidColorBrush())
            ColorNameToBrush.Add("lime", &HFF8CBF26UI.ToSolidColorBrush())
            ColorNameToBrush.Add("brown", &HFFA05000UI.ToSolidColorBrush())
            ColorNameToBrush.Add("pink", &HFFE671B8UI.ToSolidColorBrush())
            ColorNameToBrush.Add("orange", &HFFF09609UI.ToSolidColorBrush())
            ColorNameToBrush.Add("blue", &HFF1BA1E2UI.ToSolidColorBrush())
            ColorNameToBrush.Add("red", &HFFE51400UI.ToSolidColorBrush())
            ColorNameToBrush.Add("green", &HFF339933UI.ToSolidColorBrush())
            ColorNameToBrush.Add("mango", &HFFF09609UI.ToSolidColorBrush())
        End Sub

        Public Function Convert(ByVal value As Object, ByVal targetType As Type,
                                ByVal parameter As Object,
                                ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.Convert
            Dim v As String = value

            If v Is Nothing Then
                Throw New Exception("value")
            End If

            Dim brush As SolidColorBrush = Nothing
            If ColorNameToBrush.TryGetValue(v.ToLowerInvariant(), brush) Then
                Return brush
            End If

            Return Nothing
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type,
                                    ByVal parameter As Object,
                                    ByVal culture As Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException
        End Function
    End Class

End Namespace